Readme improvements

novemberborn opened the pull request:
I was going to document watch mode but then realized I hadn't really read the README in a while. I figured I'd use my fresh set of eyes to try and improve it instead.

I've tried to document rationale for each change in the individual commits. I'll squash these into a single commit before merging. Happy to wait until some of the outstanding PRs are in to avoid those having to be rebased again.

Hopefully this version of the readme has some better examples and brings various nuances and hidden behaviors to the fore. Let me know what else I should change, or revert :smile: 

P.S. Apologies to the awesome translators for this massive update. You folks rock.

jamestalmage commented:
> These should remain tab indented.

I just pushed up the `fix-readme-improvements` branch which removes the whitespace changes. I was trying to come up with a way to script that (I was unsuccessful).

sindresorhus commented:
Just some minor inline nits, but generally looks very good :)

jamestalmage commented:
It is probably a good idea to leave this branch accessible with no squashing of commits. It may be easier for translators to follow along commit by commit. We should still only push a single squashed commit to master, but lets leave the history here to help the translators.

forresst commented:
@jamestalmage 
>It is probably a good idea to leave this branch accessible with no squashing of commits. It may be easier for translators to follow along commit by commit. We should still only push a single squashed commit to master, but lets leave the history here to help the translators.

:+1: 
It's a very good idea

novemberborn commented on readme.md:
```diff
@@ -158,230 +172,265 @@ Arguments passed to the CLI will always take precedence over the configuration i
 
+`.only` applies across all test files, so if you use it in one file, no tests from the other file will run.
+
 ### Skip-tests
```
Need to retitle to `Skipping tests` to be consistent with the other titles.

novemberborn commented:
> These should remain tab indented.

Fair enough. I'll make them consistent in the other direction. Keeping spaces for JSON blocks and CLI output.

> I was trying to come up with a way to script that (I was unsuccessful).

@jamestalmage script what? I used the Whitespace package in Atom which did the trick, although the right indentation varies by code block which is annoying.

> It is probably a good idea to leave this branch accessible with no squashing of commits. It may be easier for translators to follow along commit by commit. We should still only push a single squashed commit to master, but lets leave the history here to help the translators.

Yes I was planning to squash and merge directly on `master` from my own machine and leaving this PR as is.

---

Updated. Added a bunch more changes (see commits after `reword faq`).

Timing wise perhaps best to land #592 and #573 first. It's easier to update this PR so those new doc inserts read well with the rest of the changes here than it is to update them in their respective PRs.

sindresorhus commented:
> although the right indentation varies by code block which is annoying.

Blame npm. They're the one requiring 2 space indent.

> Timing wise perhaps best to land #592 and #573 first.

:+1:

jamestalmage commented:
> script what?

A way to apply a series of commits, excluding whitespace only changes. `git` flags like `--ignore-spacing-changes` had me thinking it was doable. Turns out those flags don't do what I assumed they did

novemberborn commented:
Updated with improved `--match` documentation. @kasperlewau does it look OK to you (see last commit).

kasperlewau commented:
Had a read through and I think that looks just fine @novemberborn :+1:

vdemedes commented:
If @novemberborn made all requested changes, is it good to go or anyone else has some more suggestions?

novemberborn commented:
@vdemedes I'm waiting for #573 to land first.

sindresorhus commented:
@novemberborn #573 was landed. Can you fix the merge conflict and squash merge this to master?

novemberborn commented:
Updated. I decided to combine the ES2015 and Babel config sections, it read a lot better that way. Also reworded it a little.

@spudly are you happy with my changes? See the last commit.

I'll merge if people are :+1:

sindresorhus commented:
@novemberborn Looks good. Just merge now. We can follow up later with improvements if anyone has comments. This changes a lot of things, so would be good to get it merged asap.

sindresorhus commented:
Thanks for leveling up the docs @novemberborn :)

spudly commented:
@novemberborn, no complaints here. Looks good.
